File: project/boram-onboarding/src/main/java/inner/circle/boram_onboarding/survay/entity/Response.java
package inner.circle.boram_onboarding.survay.entity;


import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter @Setter
public class Response {
    @Id
    @GeneratedValue
    private Long id;
    @ManyToOne
    private Survey survey;
    private LocalDateTime submittedAt;
    @OneToMany(mappedBy = "response", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Answer> answers = new ArrayList<>();
}

File: project/hyewon-onboarding/src/main/kotlin/com/survey/model/Response.kt
package com.survey.model

import jakarta.persistence.*
import java.time.LocalDateTime
import java.util.*

@Entity
@Table(name = "RESPONSE")
data class Response(
    @Id
    @Column(name = "ID")
    val id: UUID = UUID.randomUUID(),

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "SURVEY_ID", nullable = false)
    val survey: Survey,

    @Column(name = "RESPONDENT")
    val respondent: String? = null,

    @Column(name = "UPDATE_DT")
    val updateDt: LocalDateTime = LocalDateTime.now(),

    @OneToMany(mappedBy = "response", cascade = [CascadeType.ALL], orphanRemoval = true)
    val responseItems: List<ResponseItem> = listOf(),

    @Column(name = "CREATE_DT")
    val createDt: LocalDateTime = LocalDateTime.now()
)
